cmd/nirvana/project/init.go

```diff
@@ -0,0 +1,782 @@
+	  go build -i -v -o $(OUTPUT_DIR)/$${target}                                       \
+	    -ldflags "-s -w -X $(ROOT)/pkg/version.Version=$(VERSION)                      \
+	    -X $(ROOT)/pkg/version.Commit=$(COMMIT)                                        \
+	    -X $(ROOT)/pkg/version.Package=$(ROOT)"                                        \
+	    $(CMD_DIR)/$${target};                                                         \
+	done
+
+container:
+	@for target in $(TARGETS); do                                                      \
+	  for registry in $(REGISTRIES); do                                                \
+	    image=$(IMAGE_PREFIX)$${target}$(IMAGE_SUFFIX);                                \
+	    docker build -t $${registry}$${image}:$(VERSION)                               \
+	      --build-arg ROOT=$(ROOT) --build-arg TARGET=$${target}                       \
+	      --build-arg CMD_DIR=$(CMD_DIR)                                               \
+	      -f $(BUILD_DIR)/$${target}/Dockerfile .;                                     \
+	  done                                                                             \
+	done
+
+push: container
+	@for target in $(TARGETS); do                                                      \
+	  for registry in $(REGISTRIES); do                                                \
+	    image=$(IMAGE_PREFIX)$${target}$(IMAGE_SUFFIX);                                \
+	    docker push $${registry}$${image}:$(VERSION);                                  \
+	  done                                                                             \
+	done
+
+test:
+	@go test ./...
+
+clean:
+	@rm -vrf ${OUTPUT_DIR}/*
+`
+}
+
+func (o *initOptions) templateProject() string {
+	return `
+{{- if .Boilerplate -}}
+{{ .SharpBoilerplate }}
+#
+{{ end -}}
+# This file is used to generate api docs.
+# All fileds in this file are descriptions and won't affect your nirvana configurations.
+
+project: {{ .ProjectName }}
+description: This project uses nirvana as API framework
+schemes:
+- http
+hosts:
+- localhost:8080
+contacts:
+- name: nobody
+  email: nobody@nobody.io
+  description: Maintain this project
+versions:
+- name: v1
+  description: The v1 version is the first version of this project
+  rules:
+  - "^/api/v1.*"
+`
+}
+
+func (o *initOptions) Manuals() string {
+	return `
+This command generates standard nirvana project structure.
+.
+├── bin
+├── build
+│   └── <project-name>
+│       └── Dockerfile
+├── cmd
+│   └── <project-name>
+│       └── main.go
+├── Gopkg.toml
+├── Makefile
+├── pkg
+│   ├── api
+│   │   ├── api.go
+│   │   ├── middlewares
+│   │   │   └── middlewares.go
+│   │   └── v1
+│   │       ├── converters
+│   │       ├── descriptors
+│   │       │   ├── descriptors.go
+│   │       │   └── message.go
+│   │       └── middlewares
+│   │           └── middlewares.go
+│   ├── message
+│   │   └── message.go
+│   └── version
+│       └── version.go
+├── project.yaml
```

Comment:
`project.yaml` is a bit confusing, how about just `config.yaml` or `nirvana.yaml`?